mapit/shortcuts.py

```diff
@@ -33,47 +40,64 @@ def default(self, o):
 def render(request, template_name, context=None):
     if context is None:
         context = {}
-#    context['base'] = base or 'base.html'
-#    context['connection'] = connection
     return render_to_response(
         template_name, context, context_instance=RequestContext(request)
     )
 
 
-def sorted_areas(areas):
-    # In here to prevent a circular dependency
-    from mapit import countries
-    if hasattr(countries, 'sorted_areas'):
-        return countries.sorted_areas(areas)
-    return list(areas)
-
-
 def output_html(request, title, areas, **kwargs):
     kwargs['json_url'] = request.get_full_path().replace('.html', '')
     kwargs['title'] = title
-    kwargs['areas'] = sorted_areas(areas)
     kwargs['indent_areas'] = kwargs.get('indent_areas', False)
-    return render(request, 'mapit/data.html', kwargs)
+    tpl = loader.render_to_string('mapit/data.html', kwargs, context_instance=RequestContext(request))
+    wraps = tpl.split('!!!DATA!!!')
+
+    def template_area(area):
+        return loader.render_to_string('mapit/areas_item.html', {'area': area, 'indent_areas': kwargs['indent_areas']})
+    areas = map(template_area, areas)
```

Comment:
This could probably be sped up by not loading/compiling the template for each area (even if template caching is being used, it'd save a round-trip to the cache). Something like this maybe?

```
    item_tpl = loader.get_template('mapit/areas_item.html')
    areas = map(lambda area: item_tpl.render(Context({'area': area})), areas)
```